test/E2ETest/Infrastructure/EqualityYamlNodeVisitor.cs

```diff
@@ -0,0 +1,136 @@
+﻿using System;
+using Tye.Serialization;
+using YamlDotNet.RepresentationModel;
+
+namespace E2ETest
+{
+    public class EqualityYamlNodeVisitor
+    {
+        public EqualityYamlNodeVisitor()
+        {
+        }
+
+        public void Visit(YamlNode node, YamlNode otherNode)
+        {
+            if (node.NodeType == YamlNodeType.Mapping)
+            {
+                VisitMapping((YamlMappingNode)node, (YamlMappingNode)otherNode);
```

Comment:
We should have a good error for the case where the cast fails on `otherNode`.

Reply:
Put another way, if `node` and `otherNode` aren't the same type, you'll get a garbo error.